plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3select/IonSqlQueryBuilder.java

```diff
@@ -252,22 +277,11 @@ private String createColumn(Type type, HiveColumnHandle columnHandle)
         String column = getFullyQualifiedColumnName(columnHandle);
 
         if (type.equals(BIGINT) || type.equals(INTEGER) || type.equals(SMALLINT) || type.equals(TINYINT)) {
-            return formatPredicate(column, "INT");
+            return "CAST(" + column + " AS INT)";
         }
         if (type.equals(BOOLEAN)) {
-            return formatPredicate(column, "BOOL");
-        }
-        if (type.equals(DATE)) {
-            return formatPredicate(column, "TIMESTAMP");
-        }
-        if (type instanceof DecimalType decimalType) {
-            return formatPredicate(column, format("DECIMAL(%s,%s)", decimalType.getPrecision(), decimalType.getScale()));
+            return "CAST(" + column + " AS BOOL)";
         }
         return column;
     }
-
-    private String formatPredicate(String column, String type)
-    {
-        return format("case %s when '' then null else CAST(%s AS %s) end", column, column, type);
```

Comment:
Why are we removing this? What happens in case the column value is actually empty String? CAST would fail on empty / 'NULL' strings.

Reply:
The check is still there, it is just up one level so it is done once per Domain instead of once per predicate.

The `case` statement was not correct, though, for JSON files. Since it has a real null type and should not encode them as empty strings.